minimal.js

```diff
@@ -144,7 +150,25 @@ module.exports = function (dirname, keys, opts) {
   }
 
   db.addBoxer = function addBoxer (boxer) {
-    boxers.push(boxer)
+    switch (typeof boxer) {
+      case 'function':
+        boxers.push(boxer)
+        break
+
+      case 'object':
+        if (typeof boxer.value !== 'function') throw new Error('invalid boxer')
+
+        if (boxer.init) {
+          setup.boxers.add(boxer.init)
```

Comment:
I think I made it so this could be a one-line `setup.boxers.add(boxer.init).runAll()` but not likely worth the change